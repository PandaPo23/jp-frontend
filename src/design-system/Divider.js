import React from 'react';

import Absolute from './Absolute';
import Flex from './Flex';
import Text from './Text';

const Divider = ({
  text,
  children,
  thickness = 1,
  length = 1,
  vertical = false,
  p = 2,
  width = 1,
  height = '100%',
  ...props
}) => (
  <Flex
    display="inline-flex"
    position="relative"
    centered
    p={p}
    width={!vertical && width}
    height={vertical && height}
    {...props}
  >
    <Absolute
      width={!vertical && 1}
      height={vertical && '100%'}
      top={!vertical && '50%'}
      left={vertical && '50%'}
      borderTop={!vertical && thickness}
      borderLeft={vertical && thickness}
    >
      &nbsp;
    </Absolute>
    {text && !children && (
      <Text zIndex={1} p={2}>
        {text}
      </Text>
    )}
    {children}
  </Flex>
);

export default Divider;
