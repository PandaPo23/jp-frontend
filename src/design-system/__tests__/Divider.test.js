import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Divider from '../Divider';
import Relative from '../Relative';

describe('Divider component', () => {
  it('renders horizontal divider correctly', () => {
    const wrapper = mount(<Divider text="OR" width={1} bg="surface" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders vertical divider correctly', () => {
    const wrapper = mount(
      <Divider borderColor="primary" vertical thickness={2} height={150}>
        <Relative p={2} borderRadius="sm" primary uppercase>
          then
        </Relative>
      </Divider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
