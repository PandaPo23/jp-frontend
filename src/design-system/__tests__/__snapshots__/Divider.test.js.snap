// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component renders horizontal divider correctly 1`] = `
.emotion-2 {
  background-color: surface;
  padding: 8px;
  width: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  width: 100%;
  border-top: 1px;
  position: absolute;
  top: 50%;
  position: absolute;
}

.emotion-1 {
  padding: 8px;
  z-index: 1;
}

<Divider
  bg="surface"
  text="OR"
  width={1}
>
  <Flex
    bg="surface"
    centered={true}
    display="inline-flex"
    height={false}
    p={2}
    position="relative"
    width={1}
  >
    <div
      className="emotion-2"
    >
      <Absolute
        borderLeft={false}
        borderTop={1}
        height={false}
        left={false}
        position="absolute"
        top="50%"
        width={1}
      >
        <div
          className="emotion-0"
        >
           
        </div>
      </Absolute>
      <Text
        p={2}
        zIndex={1}
      >
        <div
          className="emotion-1"
        >
          OR
        </div>
      </Text>
    </div>
  </Flex>
</Divider>
`;

exports[`Divider component renders vertical divider correctly 1`] = `
.emotion-2 {
  padding: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 150px;
  border-color: primary;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  height: 100%;
  border-left: 2px;
  left: 50%;
  position: absolute;
  position: absolute;
}

.emotion-1 {
  padding: 8px;
  border-radius: sm;
  position: relative;
  position: relative;
}

<Divider
  borderColor="primary"
  height={150}
  thickness={2}
  vertical={true}
>
  <Flex
    borderColor="primary"
    centered={true}
    display="inline-flex"
    height={150}
    p={2}
    position="relative"
    width={false}
  >
    <div
      className="emotion-2"
    >
      <Absolute
        borderLeft={2}
        borderTop={false}
        height="100%"
        left="50%"
        position="absolute"
        top={false}
        width={false}
      >
        <div
          className="emotion-0"
        >
           
        </div>
      </Absolute>
      <Relative
        borderRadius="sm"
        p={2}
        position="relative"
        primary={true}
        uppercase={true}
      >
        <div
          className="emotion-1"
        >
          then
        </div>
      </Relative>
    </div>
  </Flex>
</Divider>
`;
